src/odemis/util/test/img_test.py

```diff
@@ -1181,8 +1183,825 @@ def test_rotate_img_metadata(self):
+
+        # create an ordered list of (float) focus positions from 1-8um
+        self.z_list = [1e-06, 2e-06, 3e-06, 4e-06, 5e-06, 6e-06, 7e-06, 8e-06]
+
+    def test_assemble_zcube_normal(self):
+        """
+        Verify that using a 2D shaped image DA's in a list will return the right ZCube assembly.
+        """
+        img_list = []
+
+        # create a list of DA
+        for x in range(len(self.z_list)):
+            img_list.append(model.DataArray(numpy.random.randint(0, 220, self.size), self.md))
+
+        output_DA = img.assembleZCube(img_list, self.z_list)
+
+        # check if the shape of the returned DA is a combination of the number of z-slices and the size
+        input_shape = (len(img_list),) + self.size
+        self.assertEqual(output_DA.shape, input_shape)
+        # check if metadata includes the centre position of the z-stack
+        self.assertEqual(len(output_DA.metadata[model.MD_POS]), 3)
+        # check if metadata includes the pixel size of the z-stack
+        self.assertEqual(output_DA.metadata[model.MD_PIXEL_SIZE], (2e-06, 4e-05, 1e-06))
+        # check if the metadata contains the z-stack mean position
+        self.assertAlmostEqual(output_DA.metadata[model.MD_POS][-1], statistics.mean(self.z_list))
+
+    def test_assemble_zcube_3d(self):
+        """
+        Verify that using a 3D shaped image DA's in a list will fail the
+        assembly of a ZCube and see if a ValueError exception is raised.
+        """
+        # adjust the size for 3D DA
+        size = (10, 700, 700)
+
+        img_list = []
+
+        # create a list of DA
+        for x in range(len(self.z_list)):
+            img_list.append(model.DataArray(numpy.random.random_integers(0, 220, size), self.md))
+
+        with self.assertRaises(ValueError):
+            img.assembleZCube(img_list, self.z_list)
+
+    def test_assemble_zcube_z_order(self):
+        """
+        Verify that using a 3D shaped image DA's in a list will return a
+        different ZCube assembly if the order of the images is changed.
+        See if the differences with a list generated before and after reordering are picked up.
+        """
+        img_list = []
+        z_list = [0.1, 0.2, 0.3, 0.4, 0.5]
+
+        # create a list of DA
+        for x in range(len(z_list)):
+            img_list.append(model.DataArray(numpy.random.random_integers(0, 220, self.size), self.md))
+
+        output_da_before = img.assembleZCube(img_list, z_list)
+
+        # create a temporary reversed list based on img_list
+        rev_list = list(reversed(img_list))
+
+        output_da_after = img.assembleZCube(rev_list, z_list)
+
+        # check if the Z dimension has the right size
+        self.assertEqual(output_da_before.shape[0], len(z_list))
+        self.assertEqual(output_da_after.shape[0], len(z_list))
+        # check if the returned DA before order reverse is different after order reverse
+        testing.assert_array_not_equal(output_da_before, output_da_after)
```

Comment:
Please also check that passing the z in the other order will reverse the cube in Z.
Something like:
```python
# Passing Z in the reverse order should lead to a cube reversed in Z, with a positive pixel size
output_rev_z = img.assembleZCube(img_list, z_list[::-1])
self.assertGreater(output_rev_z.metadata[model.MD_PIXEL_SIZE][2], 0)
numpy.testing.assert_array_equal(output_da_after, output_rev_z)
```